### PR TITLE
SDIT-1583 Get descriptions and active flag from prison-api alert types

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/ReferenceDataService.kt [hunk 1]
```diff
@@ -46,7 +47,30 @@ class ReferenceDataService(
       aggregation(TermsAggregationBuilder(attribute.name).size(maxSearchResults).field(attribute.field))
     }
 
-  fun findAlertsReferenceData(): Map<String, List<String>> {
+  fun findAlertsReferenceData(): ReferenceDataAlertsResponse {
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/ReferenceDataService.kt:50]: Still need to put this behind an endpoint and make it cacheable.

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/ReferenceDataService.kt [hunk 2]
```diff
@@ -130,3 +154,18 @@ enum class ReferenceDataAttribute(keyword: Boolean = true, field: String? = null
     else -> name
   }
 }
+data class ReferenceDataAlertsResponse(val alertTypes: List<ReferenceDataAlertType> = emptyList())
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/ReferenceDataService.kt:157]: Need to add @Schema to the response when creating the endpoint.

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/ReferenceDataService.kt [hunk 3]
```diff
@@ -130,3 +154,18 @@ enum class ReferenceDataAttribute(keyword: Boolean = true, field: String? = null
...
+
+data class ReferenceDataAlertType(
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/ReferenceDataService.kt:159]: Not sure about these names - too long

**Review by andymarke — APPROVED** [review]
LGTM

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**